documentation/course/e2/README.md

```diff
@@ -2069,15 +2074,55 @@ TODO @jakubzehner: Dodać diagram pakietów, opis architektury i endpointy.
 
 | **Metoda** | **Endpoint**     | **Producent** | **Konsument** | **Opis**                                                                       |
 | ---------- | ---------------- | ------------- | ------------- | ------------------------------------------------------------------------------ |
-| `GET`      | `/int/v1/health` | clabbert      | —             | Sprawdzenie stanu głównego serwisu ([`M/03`](#m03-healthchecki-dla-serwisów)). |
+| `GET`      | `/int/v1/health` | Clabbert      | —             | Sprawdzenie stanu głównego serwisu ([`M/03`](#m03-healthchecki-dla-serwisów)). |
 
 ## Płatność
 
 TODO @piterek130: Dodać diagram pakietów, opis architektury i endpointy.
 
 ## Logistyka
 
-TODO @mlodybercik: Dodać diagram pakietów, opis architektury i endpointy.
+![Diagram pakietów Leprechaun](./images/package-diagram-leprechaun.drawio.svg)
+
+Część systemu obsługująca logistykę JakPrzyjade, zarządzająca trasami, przystankami, pojazdami, awariami oraz rozkładami jazdy podzielona została na trzy pakiety realizujące inne zadania w działaniu aplikacji obsługującej REST API. Każde z nich jest odpowiedzialne za integralną część działania tego podsystemu i są to pakiety odpowiedzialne za komunikację (`controllers`) odpowiadające na zapytania do systemu, za logikę biznesową (`services`), oraz za dostęp do bazy danych oraz mapowanie danych przychodzących na obiekty bazodanowe oraz odwrotnie (`data`). Jednym z dodatkowych wydzielonych pakietów jest pakiet organizujący funkcje pomocnicze (`helpers`) gotowe do ponownego użycia w innych częściach systemu.
+
+### API
+
+#### API publiczne
```

Comment:
**chore:** Please note down which endpoints support pagination and which have query params.